server/src/main/java/org/elasticsearch/action/admin/indices/close/TransportVerifyShardBeforeCloseAction.java

```diff
@@ -151,21 +162,32 @@ public void markShardCopyAsStaleIfNeeded(ShardId shardId, String allocationId, A
     public static class ShardRequest extends ReplicationRequest<ShardRequest> {
 
         private final ClusterBlock clusterBlock;
+        private final boolean phase1;
 
         ShardRequest(StreamInput in) throws IOException {
             super(in);
+            if (in.getVersion().onOrAfter(Version.V_4_3_0)) {
+                phase1 = in.readBoolean();
+            } else {
+                phase1 = false;
+            }
             this.clusterBlock = new ClusterBlock(in);
         }
 
-        public ShardRequest(ShardId shardId, ClusterBlock clusterBlock) {
+        public ShardRequest(ShardId shardId, boolean phase1, ClusterBlock clusterBlock) {
```

Comment:
There is no production code call-site where phase1 is set to true, is that intentional?

Reply:
seems like https://github.com/crate/crate/pull/10596/files/5ae6cb27d9b7c7c64180229942ca21d11c310741#diff-191de78233c863150a9abf36a805ea7cL555 had to be done in two steps, first sending requests with phase1 and once the global checkpointing is advanced we send another request with phase1 false. See https://github.com/elastic/elasticsearch/pull/43205/files#diff-6f976d9c43872a6dcd26e027a1667aaaR396-R411

I'll update it then